packages/eslint-plugin/src/rules/no-restricted-imports.ts

```diff
@@ -211,6 +212,21 @@ function getRestrictedPatterns(
   return [];
 }
 
+function shouldCreateRule(
+  baseRules: RuleListener,
+  options: Options,
+): baseRules is Exclude<RuleListener, Record<string, never>> {
+  if (!Object.keys(baseRules).length || options.length === 0) {
```

Comment:
[Nit]
Maybe use the same "is empty" logic for both? 😅 

```suggestion
  if (!Object.keys(baseRules).length || ! options.length) {
```

OR 

```suggestion
  if (Object.keys(baseRules).length === 0 || options.length === 0) {
```

Reply:
Yup, you're right, it looks much cleaner, thx!